src/content/reference/react-dom/hooks/useFormState.md

```diff
@@ -0,0 +1,291 @@
+---
+title: useFormState
+canary: true
+---
+
+<Canary>
+
+The `useFormState` Hook is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels). In addition, you need to use a framework that supports React Server Components to get the full benefit of `useFormState`.
+
+</Canary>
+
+<Intro>
+
+`useFormState` is a Hook that allows you to read the return value of the form action after a form is submitted.
+
+```js
+const [state, formAction] = useFormState(action, initalState);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `useFormState()` {/*useformstate*/}
+
+In the context of React Server Components, an *action* is a function that may be [executed when a form is submitted](/reference/react-dom/components/form). You can execute actions on the server or on the client.
```

Comment:
Hm we might want to remove "In the context of React Server Components" since form actions within forms still exist without RSC